### PR TITLE
Add mount_to arg to fakespawn command

### PR DESCRIPTION
**Pull request by mergu**
tested on 1.20.1

### CODE REVIEW

File: v1_20/src/main/java/com/denizenscript/denizen/nms/v1_20/helpers/PlayerHelperImpl.java [hunk 1]
```diff
@@ -101,6 +103,22 @@ public void deTrackEntity(Player player, Entity entity) {
...
+        int[] newPassengers = Arrays.copyOf(packet.getPassengers(), packet.getPassengers().length + 1);
+        newPassengers[packet.getPassengers().length] = fakeEntity.id;
+        try {
+            ENTITY_PASSENGERS.set(packet, newPassengers);
+        }
+        catch (IllegalAccessException e) {
+            Debug.echoError(e);
+        }
+        for (PlayerTag player : players) {
+            ((CraftPlayer) player.getPlayerEntity()).getHandle().connection.send(packet);
```
Comment (tal5) [inline v1_20/src/main/java/com/denizenscript/denizen/nms/v1_20/helpers/PlayerHelperImpl.java]: `PacketHelperImpl#send`

File: v1_20/src/main/java/com/denizenscript/denizen/nms/v1_20/helpers/PlayerHelperImpl.java [hunk 2]
```diff
@@ -101,6 +103,22 @@ public void deTrackEntity(Player player, Entity entity) {
 
     public record TrackerData(PlayerTag player, ServerEntity tracker) {}
 
+    @Override
+    public void addFakePassenger(List<PlayerTag> players, Entity vehicle, FakeEntity fakeEntity) {
```
Comment (tal5) [inline v1_20/src/main/java/com/denizenscript/denizen/nms/v1_20/helpers/PlayerHelperImpl.java]: `fakeEntity` -> `fakePassenger` or something like that

File: plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/FakeSpawnCommand.java [hunk 3]
```diff
@@ -54,79 +56,49 @@ public FakeSpawnCommand() {
...
+                                   @ArgName("location") @ArgLinear @ArgDefaultNull LocationTag location,
+                                   @ArgName("cancel") boolean cancel,
+                                   @ArgName("players") @ArgPrefixed @ArgDefaultNull @ArgSubType(PlayerTag.class) List<PlayerTag> players,
+                                   @ArgName("duration") @ArgPrefixed @ArgDefaultText("10s") DurationTag duration,
+                                   @ArgName("mount_to") @ArgPrefixed @ArgDefaultNull EntityTag vehicle) {
+        if (players == null) {
+            if (!Utilities.entryHasPlayer(scriptEntry)) {
+                throw new InvalidArgumentsRuntimeException("Must specify an online player!");
             }
+            players = Collections.singletonList(Utilities.getEntryPlayer(scriptEntry));
```
Comment (tal5) [inline plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/FakeSpawnCommand.java]: `List#of`

File: plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/FakeSpawnCommand.java [hunk 4]
```diff
@@ -54,79 +56,49 @@ public FakeSpawnCommand() {
...
-                    && arg.matches("cancel")) {
-                scriptEntry.addObject("cancel", new ElementTag(true));
-            }
-            else {
-                arg.reportUnhandled();
+    public static void autoExecute(ScriptEntry scriptEntry,
+                                   @ArgName("entity") @ArgLinear EntityTag entity,
+                                   @ArgName("location") @ArgLinear @ArgDefaultNull LocationTag location,
+                                   @ArgName("cancel") boolean cancel,
+                                   @ArgName("players") @ArgPrefixed @ArgDefaultNull @ArgSubType(PlayerTag.class) List<PlayerTag> players,
```
Comment (tal5) [inline plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/FakeSpawnCommand.java:76]: Looks like the `players` arg also allowed `to` as a prefix?

File: plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/FakeSpawnCommand.java [hunk 5]
```diff
@@ -54,79 +56,49 @@ public FakeSpawnCommand() {
...
-            }
-            else if (!scriptEntry.hasObject("cancel")
-                    && arg.matches("cancel")) {
-                scriptEntry.addObject("cancel", new ElementTag(true));
-            }
-            else {
-                arg.reportUnhandled();
+    public static void autoExecute(ScriptEntry scriptEntry,
+                                   @ArgName("entity") @ArgLinear EntityTag entity,
+                                   @ArgName("location") @ArgLinear @ArgDefaultNull LocationTag location,
```
Comment (tal5) [inline plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/FakeSpawnCommand.java]: I believe these two would need legacy out-of-order args handling? (note that `cancel` shouldn't need that as it's a boolean arg).

File: v1_20/src/main/java/com/denizenscript/denizen/nms/v1_20/helpers/PlayerHelperImpl.java [hunk 6]
```diff
@@ -76,6 +77,7 @@ public class PlayerHelperImpl extends PlayerHelper {
 
     public static final Field FLY_TICKS = ReflectionHelper.getFields(ServerGamePacketListenerImpl.class).get(ReflectionMappingsInfo.ServerGamePacketListenerImpl_aboveGroundTickCount, int.class);
     public static final Field VEHICLE_FLY_TICKS = ReflectionHelper.getFields(ServerGamePacketListenerImpl.class).get(ReflectionMappingsInfo.ServerGamePacketListenerImpl_aboveGroundVehicleTickCount, int.class);
+    public static final Field ENTITY_PASSENGERS = ReflectionHelper.getFields(ClientboundSetPassengersPacket.class).get(ReflectionMappingsInfo.ClientboundSetPassengersPacket_passengers, int[].class);
```
Comment (tal5) [inline v1_20/src/main/java/com/denizenscript/denizen/nms/v1_20/helpers/PlayerHelperImpl.java]: Would probably name this `PASSENGERS_PACKET_PASSENGERS` or something, `ENTITY_PASSENGERS` isn't really clear

**Comment by mergu** [comment]
Pushed up all the above requested changes

File: plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/FakeSpawnCommand.java [hunk 7]
```diff
@@ -3,33 +3,35 @@
...
+        addRemappedPrefixes("players", "to");
+        autoCompile();
     }
 
     // <--[command]
     // @Name FakeSpawn
-    // @Syntax fakespawn [<entity>] [<location>/cancel] (players:<player>|...) (d:<duration>{10s})
+    // @Syntax fakespawn [<entity>] [<location>/cancel] (players:<player>|...) (duration:<duration>{10s}) (mount_to:<entity>)
     // @Required 2
     // @Maximum 4
```
Comment (mcmonkey4eva) [inline plugin/src/main/java/com/denizenscript/denizen/scripts/commands/player/FakeSpawnCommand.java]: This value needs to be updated

**Pull request merged**